packages/guides/src/Nodes/SectionNode.php

```diff
@@ -5,15 +5,28 @@
 namespace phpDocumentor\Guides\Nodes;
 
 use function array_merge;
+use function array_unique;
+use function trim;
 
 /** @extends CompoundNode<Node> */
-final class SectionNode extends CompoundNode
+final class SectionNode extends CompoundNode implements TitledNode
 {
+    private string $identifier;
+    /** @var string[] */
+    private array $anchors = [];
+
     public function __construct(private readonly TitleNode $title)
     {
+        $this->identifier = $title->getId();
+
         parent::__construct();
     }
 
+    public function getTitlePlaintext(): string
```

Comment:
I think this will not work, as the title will contain markup?

Reply:
I think we should be able to derive a non-marked up plain text version of the title that can be used as link-text etc later? we do the same when we create the default identifier of the title?